aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,785 @@
+
+Which renders the following HTML:  
+
+.. code-block:: none
+
+  <p>
+    Last week: 6/30/2016 4:39:52 PM
+ </p>
+
+.. review comment: I removed "unless dictated by the calling of a method". How is that dictated? Need to explain that if we want to add it back.
+
+With the exception of the C# ``await`` keyword implicit expressions must not contain spaces. For example, you can intermingle spaces as long as the C# statement has a clear ending:
+
+.. code-block:: none
+
+  <p>@await DoSomething("hello", "world")</p>
+
+HTML containing ``@`` symbols may need to be escaped with a second ``@`` symbol. For example:
+  
+.. code-block:: none
+
+ <p>@@Username</p> 
+ 
+would render the following HTML:
+
+.. code-block:: none
+ 
+ <p>@Username</p> 
+
+.. review: Doesn't Razor treat anything that looks like an email alias as email? Not just in HTML attributes?
+
+.. _razor-email-ref:
+
+``@`` used in an email alias
+-----------------------------
+
+HTML attributes containing email addresses don’t treat the ``@`` symbol as a transition character.
+
+ ``<a href="mailto:Support@contoso.com">Support@contoso.com</a>``
+
+Consider the following Razor markup:
+
+.. code-block:: none
+
+  @{
+      var joe = new Person("Joe", 33);
+   }
+  
+  <p>Age @joe.Age</p>
+
+Predictably, the server renders ``<p>Age 33</p>``. But suppose you needed to concatenate the output to get ``Age33`` with no space between "Age" and "33". The following markup:
+
+.. code-block:: none
+
+  @{
+      var joe = new Person("Joe", 33);
+   }
+  
+  <p>Age@joe.Age</p>
+
+generates:
+
+.. code-block:: none
+
+  <p>Age@joe.Age</p>
+
+Razor is treating ``Age@joe.Age`` as an email alias. In cases like this, create an explicit expression with ``@()``:
+
+.. code-block:: none
+
+ <p>Age@(joe.Age)</p>
+ 
+Which renders
+
+.. code-block:: none
+
+  <p>Age33</p>
+  
+Explict Razor statement blocks surrounded by ``{}`` contain normal C# and therefore each C# statement must be terminated with the ``;`` character. Implict statements don't use ``;`` termination:
+
+.. literalinclude:: razor/sample/Views/Home/Contact.cshtml
+  :language: html
+  :start-after: }
+  :end-before: @* End of greeting *@ 
+ 
+Expression encoding
+-------------------
+
+Non-:dn:iface:`~Microsoft.AspNetCore.Html.IHtmlContent` content is HTML encoded. For example, the following Razor markup:
+
+.. code-block:: none
+
+  @("<div>Hello World</div>") 
+
+Renders this HTML:
+
+.. code-block:: none
+
+  &lt;div&gt;Hello World&lt;/div&gt;
+  
+Which the browser renders as:
+
+``<div>Hello World</div>)`` 
+
+:dn:cls:`~Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper` :dn:method:`~Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper.Raw` wraps the HTML markup in an :dn:cls:`~Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper` instance so that it is not encoded but rendered as HTML markup. :dn:cls:`~Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper` implements :dn:iface:`~Microsoft.AspNetCore.Html.IHtmlContent`
```

Comment:
`wraps the HTML markup in an IHtmlContent instance so that it is not encoded but rendered as HTML markup.`

The contract says it returns an `IHtmlContent` no need to dictate it's impl type.